bitcoin/src/bip32.rs

```diff
@@ -1007,29 +1407,41 @@ mod tests {
     ) {
         let mut sk = Xpriv::new_master(network, seed).unwrap();
         let mut pk = Xpub::from_priv(secp, &sk);
-
         // Check derivation convenience method for Xpriv
         assert_eq!(&sk.derive_priv(secp, &path).to_string()[..], expected_sk);
 
+        // Tries to convert the path into a normal path, if it fails, it should return an error
+        match path.to_normal() {
+            Ok(path) => {
+                assert_eq!(&pk.derive_pub(secp, &path).to_string()[..], expected_pk);
+            }
+            Err(e) => {
+                assert_eq!(e, Error::CannotDeriveFromHardenedKey);
+            }
+        }
+
         // Check derivation convenience method for Xpub, should error
         // appropriately if any ChildNumber is hardened
-        if path.0.iter().any(|cnum| cnum.is_hardened()) {
-            assert_eq!(pk.derive_pub(secp, &path), Err(Error::CannotDeriveFromHardenedKey));
-        } else {
-            assert_eq!(&pk.derive_pub(secp, &path).unwrap().to_string()[..], expected_pk);
-        }
+        //if path.0.iter().any(|cnum| cnum.is_hardened()) {
+        //    assert_eq!(pk.try_derive_pub(secp, &path), Err(Error::CannotDeriveFromHardenedKey));
+        //} else {
+        //    assert_eq!(&pk.derive_pub(secp, path.into()).to_string()[..], expected_pk);
+        //}
```

Comment:
Why this is commented?

Reply:
This is a old test piece that i left commented so we can work on test units after this pr

Reply:
Move to an issue. The codebase is not a dumpster...